bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyComponents.java

```diff
@@ -435,7 +435,10 @@ public static boolean updateSensors(ShellyThingInterface thingHandler, ShellySet
                     if (t.tmp != null) {
                         updated |= updateTempChannel(thingHandler, CHANNEL_GROUP_SENSOR, CHANNEL_SENSOR_TEMP,
                                 t.tmp.value, t.tmp.units);
-                        updated |= updateTempChannel(thingHandler, CHANNEL_GROUP_SENSOR, CHANNEL_CONTROL_SETTEMP,
+                        if (t.targetTemp.unit == null) {
+                            t.targetTemp.unit = t.tmp.units;
+                        }
```

Comment:
I'm also reopening this one.

Reply:
uff, also lost the comment, stupid me

This is related to buggy JSON format. Esp. Gen1 devices have problems here down to the level of returning incorrect JSON. Depending on device type and firmware version the are various work arounds in the code like this one. units vs. units is correct, because that matches the returned format. And yes, someone the unit is not returned at all, therefore I default to C (because most users are from Germany)

Reply:
OK, if unit is actually missing in the payload, the work-around here should be okay. Thanks for confirming that.

It seemed suspicious that you have "unit" and "units" and one of them would be missing the unit, so I just wondered if it could have been a typo, and actually they would both carry unit.